### PR TITLE
Feature/rr 913 role based consent

### PR DESCRIPTION
**Pull request by au2**
If this PR fixes a bug, you _must_ add test cases representative of the bug.

Please refer to [Tettra](https://app.tettra.co/teams/amida/pages/amida-pull-request-and-code-review-guide) for PR review guidelines.

#### How should this be manually tested?

1) POST /consent-types with a new field 'role'. It can be either 'participant' or 'clinician'.
2) [GET] /user-consent-documents. Check that either role does not get other's consent document.
3) Use boolean query parameter 'role-only' to show only role based consent documents with [GET] /user-consent-documents. This can be used after login to show unsigned role based consent.
4) You should not able to [POST] /survey-consents with a role consent type.
5) SAGE endpoints (/consents) are also updated similarly in case you want to test. I do not believe these are currently being used.

### CODE REVIEW

**Comment by au2** [comment]
@unprofessional please add an additional person to PR.

**Comment by unprofessional** [comment]
@au2 once we query for the role-only consent, we get only bare-minimum info along with the documentId.

In order to get the rest of the info, we need to query `user-consent-documents/{documentId}` -- but `clinician` can't access the endpoint. Can we give the endpoint `clinician` access, or should I be getting the other info some other way?

**Comment by au2** [comment]
i will add clinician

**Comment by au2** [comment]
I will do /user-consent-documents/{id} but you can also just use /consent-documents/{id}. The only additional info that /user-consent-documents/{id} is the signature information and if you are showing the user unsigned ones you now what that is (no signature).

**Comment by au2** [comment]
/user-consent-document/{id} now works for clinicians

File: test/survey-consent.integration.js [hunk 1]
```diff
@@ -116,23 +128,23 @@ describe('survey consent integration', () => {
...
+    // const listConsentDocumentsFn = function () {
+    //     return function listConsentDocuments() {
+    //         return rrSuperTest.get('/consent-documents', false, 200)
+    //             .then((res) => {
+    //                 const types = _.range(typeCount);
+    //                 const expected = hxConsentDocument.serversInList(types, { keepTypeId: true });
+    //                 comparator.consentDocuments(expected, res.body);
+    //             });
+    //     };
+    // };
```
[the remark this answers is not in the document]
Comment (au2) [reply test/survey-consent.integration.js]: removed

**Comment by au2** [comment]
Commented out code is removed. For some reason npm test is crashing (is that what you are using?) on trailing comma for function arguments while grunt and mocha passes. Probably depends on what version of mocha npm test runs. Anyway I removed multi line arguments for long lines now so npm test should run for you.

**Comment by jsachs** [comment]
Looks good to me, but `grunt` is bailing on `answer import-export integration list user 1, 3 answers:` which seems unrelated to these changes.

**Comment by jsachs** [comment]
Actually, same issue as the other PR. Tests are now bailing on `answer import-export integration list imported user 1, 3 answers and verify:`

**Comment by au2** [comment]
They pass for me. What is the error message?

**Comment by au2** [comment]
@unprofessional is this good to go?

**Comment by unprofessional** [comment]
Yep, it's good.

**Comment by unprofessional** [comment]
@au2 Actually, I just went through some UI testing on my side -- do we have any way to update consent roles?

**Comment by au2** [comment]
Sorry I just saw. No there is not.  Reopen the ticket or assign a new one.
I do not think it should take long.

**Pull request merged**